pkgs/tools/admin/puppet/puppet-bolt/default.nix

```diff
@@ -1,13 +1,36 @@
-{ bundlerApp, makeWrapper }:
+{ lib, bundlerApp, makeWrapper, bundlerUpdateScript, puppet-bolt, testers }:
 
 bundlerApp {
   pname = "bolt";
+
   gemdir = ./.;
   exes = [ "bolt" ];
   buildInputs = [ makeWrapper ];
 
+  gemConfig.bolt = attrs: {
+    # scripts in libexec will be executed by remote host,
+    # so shebangs should remain unchanged
+    dontPatchShebangs = true;
+  };
+
   postBuild = ''
     # Set BOLT_GEM=1 to remove warning
     wrapProgram $out/bin/bolt --set BOLT_GEM 1
   '';
+
+  passthru.tests.version = testers.testVersion {
+    package = puppet-bolt;
+    version = (import ./gemset.nix).bolt.version;
+  };
+
+  passthru.updateScript = bundlerUpdateScript "puppet-bolt";
+
+  meta = with lib; {
+    description = "Execute commands remotely over SSH and WinRM";
+    homepage = "https://github.com/puppetlabs/bolt";
+    license = licenses.asl20;
+    maintainers = with maintainers; [ uvnikita ];
+    mainProgram = "bolt";
```

Comment:
```suggestion
```
that should already be defaulted to

Reply:
Seems like without `mainProgram = "bolt";` `testVersion` tries to call `bolt-3.24.0 --version` and fails.

Reply:
bundlerApp is not a real package thats why it is not working. Will be fixed universally with #182816

In the meantime you can squash all the commits.

Reply:
Done